soh/soh/Enhancements/randomizer/randomizer.cpp

```diff
@@ -1191,81 +1191,537 @@ s16 Randomizer::GetRandomizedItemId(GetItemID ogId, s16 actorId, s16 actorParams
+        case RG_LENS_OF_TRUTH:
+            return INV_CONTENT(ITEM_LENS) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_MAGIC_BEAN:
+        case RG_MAGIC_BEAN_PACK:
+            return BEANS_BOUGHT < 10 ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_MEGATON_HAMMER:
+            return INV_CONTENT(ITEM_HAMMER) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_FIRE_ARROWS:
+            return INV_CONTENT(ITEM_ARROW_FIRE) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_ICE_ARROWS:
+            return INV_CONTENT(ITEM_ARROW_ICE) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_LIGHT_ARROWS:
+            return INV_CONTENT(ITEM_ARROW_LIGHT) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_DINS_FIRE:
+            return INV_CONTENT(ITEM_DINS_FIRE) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_FARORES_WIND:
+            return INV_CONTENT(ITEM_FARORES_WIND) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+        case RG_NAYRUS_LOVE:
+            return INV_CONTENT(ITEM_NAYRUS_LOVE) == ITEM_NONE ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+
+        // Bottles
+        case RG_EMPTY_BOTTLE:
+        case RG_BOTTLE_WITH_MILK:
+        case RG_BOTTLE_WITH_RED_POTION:
+        case RG_BOTTLE_WITH_GREEN_POTION:
+        case RG_BOTTLE_WITH_BLUE_POTION:
+        case RG_BOTTLE_WITH_FAIRY:
+        case RG_BOTTLE_WITH_FISH:
+        case RG_BOTTLE_WITH_BLUE_FIRE:
+        case RG_BOTTLE_WITH_BUGS:
+        case RG_BOTTLE_WITH_POE:
+        case RG_RUTOS_LETTER:
+        case RG_BOTTLE_WITH_BIG_POE:
+            return Inventory_HasEmptyBottleSlot() ? CAN_OBTAIN : CANT_OBTAIN_ALREADY_HAVE;
+
+        // Bottle Refills
+        case RG_MILK:
+        case RG_RED_POTION_REFILL:
+        case RG_GREEN_POTION_REFILL:
+        case RG_BLUE_POTION_REFILL:
+        case RG_BUY_FISH:
+        case RG_BUY_RED_POTION_30:
+        case RG_BUY_GREEN_POTION:
+        case RG_BUY_BLUE_POTION:
+        case RG_BUY_BLUE_FIRE:
+        case RG_BUY_BOTTLE_BUG:
+        case RG_BUY_POE:
+        case RG_BUY_FAIRYS_SPIRIT:
+        case RG_BUY_RED_POTION_40:
+        case RG_BUY_RED_POTION_50:
+            return Inventory_HasEmptyBottle() ? CAN_OBTAIN : CANT_OBTAIN_NEED_EMPTY_BOTTLE;
+
+        // Trade Items
+        // TODO: Do we want to be strict about any of this?
+        // case RG_WEIRD_EGG:
+        // case RG_ZELDAS_LETTER:
+        // case RG_POCKET_EGG:
+        // case RG_COJIRO:
+        // case RG_ODD_MUSHROOM:
+        // case RG_ODD_POTION:
+        // case RG_POACHERS_SAW:
+        // case RG_BROKEN_SWORD:
+        // case RG_PRESCRIPTION:
+        // case RG_EYEBALL_FROG:
+        // case RG_EYEDROPS:
+        // case RG_CLAIM_CHECK:
+        // case RG_PROGRESSIVE_GORONSWORD:
+        // case RG_GIANTS_KNIFE:
```

Comment:
@lilDavid any thoughts on these?

Reply:
As it is now, there's only one of each trade item in the pool if the trade quest is shuffled, and only one total if it isn't. If we want to allow there to be multiple in the pool, then we would need logic for that (and the way the trade items are handled would need to be tweaked as well), but as it is now, this is fine.